in-vehicle-digital-twin/src/digitaltwin_impl.rs

```diff
@@ -79,13 +64,16 @@ impl DigitalTwin for DigitalTwinImpl {
         request: Request<RegisterRequest>,
     ) -> Result<Response<RegisterResponse>, Status> {
         let request_inner = request.into_inner();
-        let dtdl = request_inner.dtdl;
 
-        info!("Received a register request for the DTDL:\n{}", &dtdl);
+        for entity_access_info in &request_inner.entity_access_info_list {
+            info!("Received a register request for the the entity:\n{}", entity_access_info.id);
 
-        let register_each_one_result = self.register_each_one(&dtdl);
-        if let Err(error) = register_each_one_result {
-            return Err(Status::internal(error));
+            match self.register_entity(entity_access_info.clone()) {
+                Ok(_) => {
+                    self.register_entity(entity_access_info.clone()).map_err(Status::internal)?
```

Comment:
this will register the entity twice. most likely this entire match statement should be replaced with the following:

`self.register_entity(entity_access_info.clone()).map_err(Status::internal)?;`

Reply:
Thanks.  Fixed.